### ISSUE TITLE
Expression of type "str" cannot be assigned to return type "TypeVar('AnyStr')"

### ISSUE THREAD

**Issue by tekumara** [microsoft/pyright#1225]
```python
import os
import os.path
from typing import AnyStr

def read_file(filepath: str) -> AnyStr:
    with open(os.path.expanduser(filepath)) as file:
        return file.read()
```

Errors with

```
  7:16 - error: Expression of type "str" cannot be assigned to return type "TypeVar('AnyStr')"
    Type "str" cannot be assigned to type "TypeVar('AnyStr')" (reportGeneralTypeIssues)
```

I'm running pyright 1.1.90 via the CLI. This error did not occur in pyright 1.1.87

Python 3.7.9

**Comment by erictraut** [comment]
Previous versions of Pyright didn't do proper type checking for constrained TypeVars like AnyStr. I think the current behavior is correct. Mypy also produces an error with this code.

In your sample, you're using a TypeVar in a generic function, but the TypeVar occurs only once. That means there's nothing for the TypeVar solver to "solve". Did you intend for the return type to be `str` or `Union[str, bytes]` instead?

**Comment by tekumara** [comment]
Thanks for the explanation @erictraut - looks like this in an incorrect use of `AnyStr` on my part. I'll change the return type to `str`.
